connection.go

```diff
@@ -671,10 +671,10 @@ type snowflakeArrowStreamChunkDownloader struct {
 }
 
 func (scd *snowflakeArrowStreamChunkDownloader) Location() *time.Location {
-	if scd.sc != nil {
+	if scd.sc != nil && scd.sc.cfg != nil {
 		return getCurrentLocation(scd.sc.cfg.Params)
 	}
-	return nil
+	return time.Now().Location()
```

Comment:
Similarly - looking at the code, we should never reach this line, scd.sc is always initalised...

This line is much more tricky anyway. This location may be completely different than snowflake session timezone, meaning it may break LTZ contracts.

Reply:
Ideally we should never hit this line but it seems that we are hitting it. I personally feel like this is a good temporary fix until we figure out why the connection is nil because having a panic seems worse to me but I am open to any other solutions